include/eve/module/core/regular/impl/is_less.hpp

```diff
@@ -0,0 +1,48 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+namespace eve::detail
+{
+  template<typename T, typename U, callable_options O>
+  EVE_FORCEINLINE constexpr common_logical_t<T, U> is_less_(EVE_REQUIRES(cpu_), O const& o, T a, U b) noexcept
+  {
```

Comment:
I continue not liking that we accept different types here, instead of dealing with them at the callable level. I bet there are bugs hiding in this.